mart/datamodules/fiftyone.py

```diff
@@ -0,0 +1,116 @@
+#
+# Copyright (C) 2022 Intel Corporation
+#
+# SPDX-License-Identifier: BSD-3-Clause
+#
+
+import logging
+import os
+from pathlib import Path
+from typing import Any, Callable, List, Optional, Tuple
+
+import numpy as np
+import torch
+from PIL import Image
+from torchvision.datasets import VisionDataset as VisionDataset_
+
+logger = logging.getLogger(__name__)
+try:
+    import fiftyone as fo
+    import fiftyone.utils.coco as fouc
+    import fiftyone.zoo as foz
+except ImportError:
+    logger.debug("fiftyone module is not installed!")
+
+__all__ = ["FiftyOneDataset"]
+
+
+class FiftyOneDataset(VisionDataset_):
+    # Adapted from FiftyOne example: https://github.com/voxel51/fiftyone-examples/blob/master/examples/pytorch_detection_training.ipynb
+
+    def __init__(
+        self,
+        dataset_name: str,
+        gt_field: str,
+        sample_tags: List[str] = [],
+        label_tags: List[str] = [],
+        transform: Optional[Callable] = None,
+        target_transform: Optional[Callable] = None,
+        transforms: Optional[Callable] = None,
+        *args,
+        **kwargs,
+    ) -> None:
+        super().__init__("", transforms, transform, target_transform)
+
+        self.gt_field = gt_field
+
+        # load FiftyOne dataset
+        if dataset_name in fo.list_datasets():
+            self.dataset = fo.load_dataset(dataset_name)
+        elif dataset_name in foz.list_zoo_datasets():
```

Comment:
I don't think we want to load zoo datasets here, because it tries to index the dataset every time.

Maybe we should revert the changes, and use the FiftyOne command line to pre-load zoo datasets as datasets. For example, `fiftyone zoo datasets load -s validation -n coco-2017-validation coco-2017` loads the validation split of coco-2017 as a dataset named "coco-2017-validation".

Reply:
ok, makes sense